qctrlopencontrols/driven_controls/predefined.py

```diff
@@ -1206,3 +1207,128 @@ def new_modulated_gaussian_control(
         detunings=np.zeros(segment_count),
         durations=np.array([segment_duration] * segment_count),
     )
+
+
+def new_drag_control(
+    rabi_rotation: float,
+    segment_count: int,
+    duration: float,
+    width: float,
+    beta: float,
+    name: Optional[str] = None,
+) -> DrivenControl:
+    r"""
+    Generates a Gaussian driven control sequence with DRAG correction applied.
+
+    The addition of DRAG further reduces leakage out of the qubit subspace via an additional
+    off-quadrature corrective driving term proportional to the derivative of the Gaussian pulse.
+
+    Parameters
+    ----------
+    rabi_rotation : float
+        Total Rabi rotation :math:`\theta` to be performed by the driven control.
+    segment_count : int
+        Number of segments in the control sequence.
+    duration : float
+        Total duration :math:`t_g` of the control sequence.
+    width : float
+        Width (standard deviation) :math:`\sigma` of the ideal Gaussian pulse.
+    beta : float
+        Amplitude scaling :math:`\beta` of the Gaussian derivative.
+    name : str, optional
+        An optional string to name the control. Defaults to ``None``.
+
+    Returns
+    -------
+    DrivenControl
+        A control sequence as an instance of DrivenControl.
+
+    See Also
+    --------
+    new_gaussian_control
+
+    Notes
+    -----
+    A DRAG-corrected (Derivative Removal by Adiabatic Gate) Gaussian driven control [#]_
+    applies a Hamiltonian consisting of a piecewise constant approximation to an ideal
+    Gaussian pulse controlling :math:`\sigma_x` while its derivative controls the
+    application of the :math:`\sigma_y` operator:
+
+    .. math::
+        H(t) = \frac{1}{2}(\Omega_G(t) \sigma_x + \beta \dot{\Omega}_G(t) \sigma_y)
+
+    where :math:`\Omega_G(t)` is simply given by :doc:`new_gaussian_control`. Optimally,
+    :math:`\beta = -\frac{\lambda_1^2}{4\Delta_2}` where :math:`\Delta_2` is the
+    anharmonicity of the system and :math:`\lambda_1` is the relative strength required
+    to drive a transition :math:`\lvert 1 \rangle \rightarrow \lvert 2 \rangle` vs.
+    :math:`\lvert 0 \rangle \rightarrow \lvert 1 \rangle`. Note
+    that this choice of :math:`\beta`, sometimes called "simple drag" or "half derivative",
+    is a first-order version of DRAG, and it excludes an additional detuning corrective term.
```

Comment:
I am referring to the fact that this implementation uses Eq. 4.34 in that 2011 paper rather than, say, Eq. 4.36 in which the detuning correction is non-zero.

Reply:
Thanks, make sense to me now.